pkg/policy/selectorcache_selector.go

```diff
@@ -111,24 +127,29 @@ func (i *identitySelector) Equal(b *identitySelector) bool {
 // of the selections. If the old version is returned, the user is
 // guaranteed to receive a notification including the update.
 func (i *identitySelector) GetSelections() identity.NumericIdentitySlice {
-	return i.selections.At(versioned.Latest())
+	readTxn := i.selectorCache.GetReadTxn()
+	defer readTxn.Close()
+	return i.GetSelectionsAt(readTxn)
 }
 
 // GetSelectionsAt returns the set of numeric identities currently
 // selected.  The cached selections can be concurrently updated. In
 // that case GetSelectionsAt() will return either the old or new version
 // of the selections. If the old version is returned, the user is
 // guaranteed to receive a notification including the update.
-func (i *identitySelector) GetSelectionsAt(version *versioned.VersionHandle) identity.NumericIdentitySlice {
-	if !version.IsValid() {
-		i.logger.Error(
-			"GetSelections: Invalid VersionHandle finds nothing",
-			logfields.Version, version,
+func (i *identitySelector) GetSelectionsAt(txn SelectorReadTxn) identity.NumericIdentitySlice {
+	if !txn.IsOpen() {
```

Comment:
is this `IsOpen` thing needed? I'm wondering as `SelectorReadTxn` is passed around as value to bunch of places so I'm not sure if `txn.open` gets actually set to false correctly.